src/main/java/flab/gotable/controller/MemberAPIController.java

```diff
@@ -0,0 +1,47 @@
+package flab.gotable.controller;
+
+import flab.gotable.dto.MemberResponse;
+import flab.gotable.dto.request.MemberSignUpRequestDto;
+import flab.gotable.exception.DuplicatedIdException;
+import flab.gotable.exception.ErrorCode;
+import flab.gotable.service.MemberService;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequiredArgsConstructor
+@Slf4j
+@RequestMapping("/members")
+public class MemberAPIController {
+
+    private final MemberService memberService;
+
+    @PostMapping("/signup")
+    public ResponseEntity<MemberResponse> signup(@RequestBody @Valid MemberSignUpRequestDto memberSignUpRequestDto) {
+        boolean isDuplicated = memberService.isDuplicatedId(memberSignUpRequestDto.getId());
+
+        if (isDuplicated) {
+            throw new DuplicatedIdException(ErrorCode.DUPLICATED_ID, ErrorCode.DUPLICATED_ID.getMessage());
+        }
```

Comment:
딱 한번만 사용하는 변수는 선언하지 않는 것이 가독성에 좋습니다.
변수가 늘어날수록 개발자가 트래킹해야할 대상이 많아지고 메서드 읽는 것이 어려워질 수 있으니 아래와 같이 활용해주세요~

```suggestion
        if (memberService.isDuplicatedId(memberSignUpRequestDto.getId())) {
            throw new DuplicatedIdException(ErrorCode.DUPLICATED_ID, ErrorCode.DUPLICATED_ID.getMessage());
        }
```